framework/packages/abstract-client/src/interchain/mod.rs

```diff
@@ -0,0 +1,75 @@
+pub(crate) mod remote_account;
+mod remote_application;
+use std::collections::HashMap;
+
+use cw_orch_interchain::{IbcQueryHandler, InterchainEnv, InterchainError};
+pub use remote_account::RemoteAccount;
+pub use remote_application::RemoteApplication;
+
+use crate::{client::AbstractClientResult, AbstractClient, Environment};
+
+// TODO: Why are we not returning ibc tx analysis after await
+/// IbcTxAnalysis after waiting for interchain action
+pub struct IbcTxAnalysisV2<Chain: cw_orch::environment::CwEnv>(
```

Comment:
why is this a new type?

Reply:
`IbcTxAnalysis` raises a warning for being `must_use`, but it's checked analysis

Reply:
@Kayanski

Reply:
I discovered that 2 days ago yes, I don't really like this because this nesting is just there to mitigate the must_use. 
I don't have a solution right now if people want this result, they should use it. If they don't want it, they should not care, and we should not return it. Don't knoow what the devX looks like. But V2 seems weird that's for sure

Reply:
Yeah there's something that we should improve on the design of that type.